src/components/ComboBox/ComboBox.tsx

```diff
@@ -0,0 +1,132 @@
+import * as React from 'react';
+import { themr, ThemedComponentClass } from 'react-css-themr';
+import { createUniqueIDFactory } from '@shopify/javascript-utilities/other';
+import { COMBOBOX } from '../ThemeIdentifiers';
+import ComboBoxItem from './ComboBoxItem';
+import Icon from '../Icon';
+import TextField from '../TextField';
+import Popover from '../Popover';
+import arrowSvg from './icons/arrow.svg';
+import * as baseTheme from './ComboBox.scss';
+
+export type Mode = 'collapsible' | 'multiple';
+
+export interface ComboBoxProps {
+  type? : string;
+  key?: string;
+  value: any;
+}
+
+export interface Props {
+  items: ComboBoxProps[];
+  label: string;
+  style?:any;
+}
+
+interface State {
+  open: boolean;
+  items: ComboBoxProps[];
+  initialItems: ComboBoxProps[];
```

Comment:
@ankitsaraogi What's the difference between `items` and `initialItems`?  Can you add comments for all properties and state as well?